packages/core/echo/echo-db/src/client/index-query-source-provider.ts

```diff
@@ -129,46 +130,38 @@ export class IndexQuerySource implements QuerySource {
         const ctx = new Context();
         currentCtx = ctx;
 
-        const singleResultProcessingTimeout = options?.timeout
-          ? { timeout: (options.timeout - (Date.now() - start)) * 0.9 }
-          : undefined;
-
         log('queryIndex raw results', {
           queryId,
-          rowLoadTimeTimeout: singleResultProcessingTimeout,
           length: response.results?.length ?? 0,
         });
 
-        let failures = 0;
-        const fetchedFromIndexCount = response.results?.length ?? 0;
-        const processedResults: Array<QueryResult | null> =
-          fetchedFromIndexCount > 0
-            ? await Promise.all(
-                response.results!.map((result) =>
-                  this._filterMapResult(ctx, start, result, singleResultProcessingTimeout).catch((err) => {
-                    failures++;
-                    return null;
-                  }),
-                ),
-              )
-            : [];
-        const results = processedResults.filter(nonNullable);
-
-        log('queryIndex processed results', {
-          queryId,
-          fetchedFromIndex: fetchedFromIndexCount,
-          failed: failures,
-          loaded: results.length,
-        });
-
-        if (currentCtx === ctx) {
-          onResult(results);
-        } else {
-          log.warn('results from the previous update are ignored', { queryId });
+        try {
+          const fetchedFromIndexCount = response.results?.length ?? 0;
+          const processedResults: Array<QueryResult | null> =
+            fetchedFromIndexCount > 0
```

Comment:
this branch seems extraneous -- if `response.results.length` is 0 then `.map` will return an empty array anyway

Reply:
yeah, can be replaced with `Promise.all((response ?? []).map(...))`